includes/device-groups.inc.php

```diff
@@ -140,6 +140,10 @@ function GenGroupSQL($pattern, $search = '', $extra = 0)
                     $tables[] = $tmp;
                 }
             }
+              // patch  if  first table is locations
+               if ($tables[0]=='locations') {
+                $tables=array_reverse($tables);
```

Comment:
This will only fix some problems.  $tables is user generated, so 'devices' might not be the last field.

Reply:
murrant, but I cannot say the opposite thing but the issue on SQL query even written on discord is "SELECT DISTINCT(locations.device_id)" and on where part of line 162 of file.